### PR TITLE
refactor(core): Setup decorator based RBAC (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

File: packages/cli/src/controllers/users.controller.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ import type {
 	UserRepository,
 } from '@db/repositories';
 
+@Authorized(['global', 'owner'])
```
Comment (krynble) [inline packages/cli/src/controllers/users.controller.ts:44]: The `GET` operation in this controller must be public as all users should be able to list other users in order to display the sharing list.

**Review by krynble — CHANGES_REQUESTED** [review]
Did some manual testing and could only find this issue but it was blocking the very first screen after logging in.

File: packages/cli/src/controllers/users.controller.ts (hunk 1 above)
Comment (netroy) [reply packages/cli/src/controllers/users.controller.ts:44]: great catch. fixed, and added a test.

**Review by krynble — APPROVED** [review]
Ran some more tests and could not find any issues. LGTM.

**Comment by janober** [comment]
Got released with [`n8n@0.226.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.226.1)

**Pull request merged**